sklearn/cluster/_affinity_propagation.py

```diff
@@ -178,6 +178,12 @@ def _affinity_propagation(
 # Public API
 
 
+@validate_params(
+    {
+        "S": ["array-like"],
+        "return_n_iter": ["boolean"],
+    }
+)
 def affinity_propagation(
     S,
     *,
```

Comment:
The test is going to fail because we don't delegate all parameter validation to the class. We need to change the code below from

```python
    S = as_float_array(S, copy=copy)

    estimator = AffinityPropagation(
        damping=damping,
        max_iter=max_iter,
        convergence_iter=convergence_iter,
        copy=False,
        preference=preference,
        affinity="precomputed",
        verbose=verbose,
        random_state=random_state,
    ).fit(S)
```

to

```python
    estimator = AffinityPropagation(
        damping=damping,
        max_iter=max_iter,
        convergence_iter=convergence_iter,
        copy=copy,
        preference=preference,
        affinity="precomputed",
        verbose=verbose,
        random_state=random_state,
    ).fit(S)
```

So, we should not validate `S` and pass the parameter `copy` that will be validated by the class.